src/main/java/com/health/threadpool/TaskProcessingService.java

```diff
@@ -53,9 +65,319 @@ public class TaskProcessingService {
+                tutorial.setAddedQueue(true);
+
+        }
+
+        else if (tutorial.isAddedQueue()) {
+            if (requestType.equals(CommonData.UPDATE_DOCUMENT) || requestType.equals(CommonData.UPDATE_DOCUMENT_RANK)
+                    || requestType.equals(CommonData.DELETE_DOCUMENT)) {
+                Map<String, String> resultMap = addDocument(documentId, documentType, documentPath, documentUrl, rank,
+                        view_url, languageId, languag, categoryId, category, topicId, topic, outlinePath, requestType);
+
+                if (requestType.equals(CommonData.DELETE_DOCUMENT)) {
+                    if (resultMap.containsValue(CommonData.SUCCESS))
+                        tutorial.setAddedQueue(false);
+
+                }
+            }
+
+        }
+
+        tutService.save(tutorial);
+    }
+
+    public void addUpdateDeleteResearchPaper(ResearchPaper researchPaper, String requestType) {
+
+        String documentId = CommonData.DOCUMENT_TYPE_RESEARCHPAPER + researchPaper.getId();
+        String documentType = CommonData.DOCUMENT_TYPE_RESEARCHPAPER;
+        String documentPath = researchPaper.getResearchPaperPath();
+        String documentUrl = "/ResearchPaper/ " + researchPaper.getId();
+        int rank = 5 * researchPaper.getResearchPaperVisit();
+        String view_url = "test";
+        int languageId = 22;
+        String languag = "English";
+
+        if (requestType.equals(CommonData.ADD_DOCUMENT) && !researchPaper.isAddedQueue()) {
+
+            Map<String, String> resultMap = addDocument(documentId, documentType, documentPath, documentUrl, rank,
+                    view_url, languageId, languag, null, null, null, null, null, requestType);
+            if (resultMap.containsValue(CommonData.SUCCESS))
+                researchPaper.setAddedQueue(true);
+
+        }
+
+        else if (researchPaper.isAddedQueue()) {
+            if (requestType.equals(CommonData.UPDATE_DOCUMENT) || requestType.equals(CommonData.UPDATE_DOCUMENT_RANK)
+                    || requestType.equals(CommonData.DELETE_DOCUMENT)) {
+                Map<String, String> resultMap = addDocument(documentId, documentType, documentPath, documentUrl, rank,
+                        view_url, languageId, languag, null, null, null, null, null, requestType);
+
+                if (requestType.equals(CommonData.DELETE_DOCUMENT)) {
+                    if (resultMap.containsValue(CommonData.SUCCESS))
+                        researchPaper.setAddedQueue(false);
+
+                }
+            }
+
+        }
+
+        researchPaperService.save(researchPaper);
+    }
+
+    public void addUpdateDeleteBrochure(Brouchure brochure, String requestType) {
+
+        Version version = verRepository.findByBrouchureAndBroVersion(brochure, brochure.getPrimaryVersion());
+        List<FilesofBrouchure> filesOfBroList = filesofBroService.findByVersion(version);
+
+        int rank = 5 * brochure.getBrochureVisit();
+        String view_url = null;
```

Comment:
```suggestion
        String view_url = brochure.getPosterPath();
```
Change to URL of the brochure.